internal/tasks/templateversion_populate.go

```diff
@@ -110,23 +110,23 @@ func (t *TemplateVersionPopulateLogic) getFleetAndTemplateVersion(ctx context.Co
 	return nil
 }
 
-func (t *TemplateVersionPopulateLogic) handleConfigItem(ctx context.Context, configItem *api.DeviceSpec_Config_Item) error {
-	disc, err := configItem.Discriminator()
+func (t *TemplateVersionPopulateLogic) handleConfigItem(ctx context.Context, configItem *api.ConfigProviderSpec) error {
+	configType, err := configItem.Type()
 	if err != nil {
-		return fmt.Errorf("failed getting discriminator: %w", err)
+		return fmt.Errorf("failed getting config type: %w", err)
 	}
 
-	switch disc {
-	case string(api.TemplateDiscriminatorGitConfig):
```

Comment:
Nit: The types `TemplateDiscriminator`, `TemplateDiscriminatorGitConfig` etc can be removed.
They are defined in the API and in the openapi spec.

Reply:
Removed, thank you.